graphql/api/src/main/resources/stream-registry.graphql

```diff
@@ -33,24 +33,52 @@ type Tag {
 
 ############ Specification ############
 
+input PrincipalInput {
+    name: String
+}
+
+input SecurityInput {
+    role: String
+    principals: [PrincipalInput!]!
+}
+
+input SecurityQuery {
+    roleRegex: String!
+    principalRegex: String!
+}
+
+type Security {
```

Comment:
In the graphql layer, I think it makes sense to keep the raw strings, otherwise it's adding unnecessary bloat for the API.

i.e.

```
type Security {
  role: String!
  principals: [String!]!
}
```

I realise we will hit the same problem that we did for List vs Map. This will affect the model data structures and it'll have to be `Map<String, List<String>>` in the service model but `Map<Role, List<Principal>>` in lower layers. @teabot comments?

Reply:
This is true, though currently the `Principal` is just modelled as a `@Value` type. We'd also need to check on the state of graphql input union types or `OneOf` types - quick google later - still open but making progress - it's a rabbit hole of competing PRs https://github.com/graphql/graphql-spec/issues/488

Reply:
maybe at the graphql layer we separate out to something like:

```
type Security {
  role: String!
  users: [String!]!
  groups: [String!]!
}
```

but collapse it down to `Map<Role, List<Principal>>`.

Reply:
@nahguam - I talked with @teabot and we decided to leave the Principal as just a Value instead of a more complex type to avoid schema/enum complexity issues down the road. I have made the change you original suggested to keep the input Principals as a list of strings. I like this change also it should make the graphql simpler.